test_suite/cloud/test_aws.py

```diff
@@ -558,8 +558,24 @@ def test_yum_group_install(self, host):
             pytest.skip('Not applicable to Atomic host AMIs')
 
         with host.sudo():
-            assert host.run_test('yum -y groupinstall "Development tools"'), \
-                'Error while installing Development tools group'
+            dev_tools_install_command = 'yum -y groupinstall "Development tools"'
+            result = host.run(dev_tools_install_command)
+
+            if result.failed:
+                print(f'Command faild with error on first attempt: {result.stderr}')
+                err_message = "This system is not registered to Red Hat Subscription Management"
+                if err_message in result.stderr:
+                    print('"Development tools" installation attempt failed. Trying to apply a workaround...')
+                    host.run(
+                        'echo -e "enabled=0" > /etc/yum/pluginconf.d/subscription-manager.conf'
+                        ' && yum clean all'
+                    )
+
+                    assert host.run(dev_tools_install_command).succeeded, (
+                        f'Error while installing Development tools '
+                        f'group after two attempts with error: {result.stderr}'
+                    )
+            print('"Development tools" installed successfully.')
```

Comment:
I think there's still an indentation issue here. We want to print this message even if installation succeeded after the first attempt (without having to enter into the if for a second attempt, applying workaround, etc.).

```suggestion
        print('"Development tools" installed successfully.')
```

So, by doing this, we will make sue that the `"Development tools" installed successfully.` message will be printed in all the "successful" scenarios (that is, with and without a retry).

Reply:
I don't see the issue you mentioned here, according to the existing code it does what you said.
It's outside the 'if' statement, so if that's not true (pass)it will be printed
and if it passes the assert inside this "if" it will be printed.

Reply:
Yeah, forget about what I said, it looks like the comment was added while you were pushing again, so it does not make sense any longer. Let's try to be more careful next time to avoid this situation where comments can be outdated with multiple pushes done in a short amount of time.